cadquery/cq.py

```diff
@@ -3441,48 +3478,133 @@ def loft(
+
         :return: OCCT solid(s), suitable for boolean operations.
 
         This method is a utility method, primarily for plugin and internal use.
         It is the basis for cutBlind, extrude, cutThruAll, and all similar methods.
         """
 
+        def getFacesList(eDir, additive, both=False):
+            """
+            Utility function to make the code further below more clean and tidy
+            Performs some test and raise appropriate error when no Faces are found for extrusion
+            """
+            if additive:
+                direction = "AlongAxis"
+            else:
+                direction = "Opposite"
+
+            facesList = self.findSolid().facesIntersectedByLine(
+                ws[0].Center(), eDir, direction=direction
+            )
+            if len(facesList) == 0 and both:
+                raise ValueError(
+                    "Couldn't find a face to extrude/cut to for at least one of the two required directions of extrusion/cut."
+                )
+
+            if len(facesList) == 0:
+                # if we don't find faces in the workplane normal direction we try the other direction (as the user might have created a workplane with wrong orientation)
+                facesList = self.findSolid().facesIntersectedByLine(
+                    ws[0].Center(), eDir.multiply(-1.0), direction=direction
+                )
+                if len(facesList) == 0:
+                    raise ValueError(
+                        "Couldn't find a face to extrude/cut to. Check your workplane orientation."
+                    )
+            return facesList
+
         # group wires together into faces based on which ones are inside the others
         # result is a list of lists
 
         wireSets = sortWiresByBuildOrder(self.ctx.popPendingWires())
 
         # compute extrusion vector and extrude
-        eDir = self.plane.zDir.multiply(distance)
+        if upToFace is not None:
+            eDir = self.plane.zDir
+        elif distance is not None:
+            eDir = self.plane.zDir.multiply(distance)
+
+        if additive:
+            direction = "AlongAxis"
+        else:
+            direction = "Opposite"
 
         # one would think that fusing faces into a compound and then extruding would work,
-        # but it doesn't-- the resulting compound appears to look right, ( right number of faces, etc)
+        # but it doesnt-- the resulting compound appears to look right, ( right number of faces, etc)
         # but then cutting it from the main solid fails with BRep_NotDone.
         # the work around is to extrude each and then join the resulting solids, which seems to work
 
         # underlying cad kernel can only handle simple bosses-- we'll aggregate them if there are
         # multiple sets
+        thisObj: Union[Solid, Compound]
 
         toFuse = []
+        taper = 0.0 if taper is None else taper
+        baseSolid = None
```

Comment:
```suggestion
```

As far as I can tell this is always overridden by line 3562.

Reply:
No the purpose of the `baseSolid = None` is to handle what the function will do when extruding several wires.

At the first iteration (and if there is only one pending wire) the function will extrude the wire using the solid found in the Workplane with `findSolid`.

If there is several wires, at each iteration the "baseSolid" passed to the `dprism` function is the result of the previous extrusion with the previous wire.